API/Configuration/GlobalExceptionHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System.Net;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Configuration;
+
+public class GlobalExceptionHandler : IExceptionHandler
+{
+    private readonly ILogger<GlobalExceptionHandler> _logger;
+    public GlobalExceptionHandler(ILogger<GlobalExceptionHandler> logger)
+    {
+        _logger = logger;
+    }
+    public async ValueTask<bool> TryHandleAsync(HttpContext httpContext, Exception exception, CancellationToken cancellationToken)
+    {
+        var internalServerError = HttpStatusCode.InternalServerError;
+        var path = httpContext.Request.Path;
+        ProblemDetails errorResponse = new ProblemDetails();
+        _logger.LogError("Exception happend at {path} for connection id: {id}, exception trace: {type}  Message: {exception}.", path, httpContext.Connection.Id, exception.StackTrace, exception.Message);
+        switch (exception)
+        {
+            default:
+                errorResponse.Type = "https://datatracker.ietf.org/doc/html/rfc9110#section-15.6.1";
```

Comment:
How does this work? This link sends me to a site explaining Internal Server Error, but it the URL converted to a type, or is the URL somehow served to the frontend?

Reply:
The type parameter is standard for the problemdetail object created by ASP.NET, we can remove it, I added it for the reason of it being automatically added if a "real" exception was thrown without the middleware.